### ISSUE TITLE
jwalk hangs on single core computers

### ISSUE THREAD

**Issue by tavianator** [jessegrosjean/jwalk#15]
First noticed this in https://github.com/bootandy/dust/issues/52

If the rayon global pool has only one thread, e.g. because you're on a one-core computer, then jwalk will hang in its default configuration.

```
$ RAYON_NUM_THREADS=1 cargo test
...

     Running target/debug/deps/mod-d1dd66d117582654

running 14 tests
test error_when_path_does_not_exist ... ok
test error_when_path_removed_durring_iteration ... ok
test filter_group_children_with_process_entries ... ^C
```

The last test never completes.

**Comment by bootandy** [comment]
From some hacky testing I think the bug is inside: ReadDirIter: 

If you force the code into the if 'Serial' branch the tests will complete

**Comment by bootandy** [comment]
Digging deeper it seems like: 

class: `OrderedQueueIter `
method: `next_relaxed`
problem: `self.pending_count())` is never being taken down to 0 on single threads. Hence the loop there doesn't terminate.

**Comment by jessegrosjean** [comment]
I think this is fixed in 0.5.0
